app/lib/summary_stats_utils.rb

```diff
@@ -50,10 +50,12 @@ def self.weekly_returning_users
   # returns a list of all missing files for entire portal for use in nightly_server_report
   def self.storage_sanity_check
     missing_files = []
-    Study.where(queued_for_deletion: false, detached: false).each do |study|
+    valid_accessions = Study.where(queued_for_deletion: false, detached: false).pluck(:accession)
```

Comment:
When not writing to DB, is there any benefit of directly iterating over an open `where` clause like before, rather than taking this newly refined approach of assigning results straightaway and thus more quickly closing the cursor?

I ask because,  if not, I imagine this is a lurking issue present in lots of other places throughout the codebase.

Reply:
The `where` clause is more performant, when not dealing with a giant amount of data.  It loads data in a single query rather than the individual ones like this update does.  The issue with `where` is the block - long running operations risk timing out.  Given that we haven't seen a ton of this error in Sentry, my guess is that it's not a large issue.

Reply:
Yup, this pluck-and-then-query-individually approach would not be appropriate in any user-facing part of our application -- it's going to generate 1 query per study, i.e. ~1500 separate, sequential queries to the database.

Reply:
Thank you both for the insight!